service_configuration_lib/spark_config.py

```diff
@@ -135,6 +137,15 @@ def get_aws_credentials(
         with open(aws_credentials_json, 'r') as f:
             creds = json.load(f)
         return (creds.get('accessKeyId'), creds.get('secretAccessKey'), None)
+    elif assume_web_identity:
+        credentials = assume_aws_role_web_identity()
+        if credentials:
+            return (credentials['AccessKeyId'], credentials['SecretAccessKey'], credentials['SessionToken'])
+        else:
+            log.warning(
+                'Tried to assume role with web identity but either '
```

Comment:
nit:
```suggestion
                'Tried to assume role with web identity but neither '
```

Reply:
*Ultra Pedantry Warning* If one was set but the other wasn't "neither" would be wrong.

Reply:
oh, i also just re-read this and saw a `not` that I didn't see before - so this is actually fine as-is and you can ignore me :p

Reply:
"Do you want your warning to be grammatically perfect or intuitive?"
"Yes"